### DISCUSSION TITLE
Twitter authentication is not working, I have tried OAuth 1 and OAuth 2

### DISCUSSION THREAD

**Discussion by LetsDevDev** [nextauthjs/next-auth#5757] [Help]
### Question 💬

I have now tried everything for 2 days.

I have followed 3 different tutorials,
I have downloaded the examples and added my variables,
All other providers work fine.

OAuth1 gives me the message "Try another account" and does not leave the sign in page in my app.
OAuth2 Redirects to Twitter and shows an error message "You weren’t able to give access to the App. Go back and try logging in again."

Is the Auth through Twitter working for anybody?

Does anyone have any good suggestions of alternative ways to auth with Twitter if this does not work?
My application is Twitter-based so I really need it to be able to auth through Twitter.

### How to reproduce ☕️

Download the examples and add your twitter keys.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
hi, it's even deployed here https://next-auth-example.vercel.app/ there's likely a configuration error somewhere in your code/app config

**Comment by eugeneZolotarenko** [comment]
I have exactly the same problem, can't figure out what's wrong.

**Comment by eugeneZolotarenko** [comment]
Ok, I figured it out. Make sure you use the credentials under "OAuth 2.0 Client ID and Client Secret" for the 2.0 version.
